src/__tests__/MissionsComponent.test.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable max-len */
+import React from 'react';
+import { render } from '@testing-library/react';
+import { useDispatch, useSelector } from 'react-redux';
+import MissionsComponent from '../components/Missions/MissionsComponent';
+import { fetchMissions } from '../Redux/Missions/missionsSlice';
+
+// Mock useSelector and useDispatch
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useDispatch: jest.fn(),
+  useSelector: jest.fn(),
+}));
```

Comment:
- [ ] The restructuring of your project files and folders has caused some test suites to fail, kindly fix them

Before Restructure
![image](https://github.com/AbuTalha3/react_group_project/assets/81932478/de66a36f-38d5-44b6-b6c2-e4c16bf0a341)

After Restructure
![image](https://github.com/AbuTalha3/react_group_project/assets/81932478/0981d20a-1521-4c80-9bb2-77561197eb57)

![image](https://github.com/AbuTalha3/react_group_project/assets/81932478/30393151-4862-44a3-86a5-c8eda569b6eb)